lib/manageiq/api/common/graphql/associated_records.rb

```diff
@@ -0,0 +1,44 @@
+require "query_relation"
+
+module ManageIQ
+  module API
+    module Common
+      module GraphQL
+        class AssociatedRecords
+          include Enumerable
+          include QueryRelation::Queryable
+
+          def initialize(collection = [])
```

Comment:
Is there a case where this would be called without a collection?

Reply:
Actually... Is there a case where this wouldn't be an `ActiveRecord::Associations::CollectionProxy`?  With that in mind, should the `filter_collection_by_options` below use the database methods rather than doing everything in ruby?

Reply:
the N+1 problem come back when I do that. 😞

Reply:
But, isn't it the same problem, just being handled in ruby instead of postgres?  It there are 10,000 subcollection items and the request has `limit: 5`, aren't we still going to instantiate 10,000 ruby objects, filter all 10,000, sort them then only return 5?